include/deal.II/grid/grid_tools.h

```diff
@@ -1285,6 +1389,272 @@ namespace GridTools
       const bool                                             perform_handshake,
       const bool enforce_unique_mapping = false);
 
+
+    /**
+     * Data structure returned by
+     * GridTools::internal::distributed_compute_intersection_locations(). It can
+     *     be converted to
+     * GridTools::internal::DistributedComputePointLocationsInternal, which can
+     * be used to reinit Utilities::MPI::RemotePointEvaluation.
+     */
+    template <int dim, int spacedim, typename IntersectionRequest>
+    struct DistributedComputeIntersectionLocationsInternal
+    {
+      // Intersection answers are assumed to be simplices (e.g. provided by
+      // CGAL)
+      using IntersectionAnswer = std::array<dealii::Point<dim>, dim>;
+
+      /**
+       * Information of each intersection on sending/evaluation side. The
+       * elements of the tuple are as follows: 0) cell level and index, 1) rank
+       * of the owning process, 2) local index of the owning process, 3)
+       * found intersection, 4) requested intersection.
+       *
+       * @note The vector is sorted according to 1), 2).
+       */
+      std::vector<std::tuple<std::pair<int, int>,
+                             unsigned int,
+                             unsigned int,
+                             IntersectionAnswer,
+                             IntersectionRequest>>
```

Comment:
Is this last entry of the tuple actually needed!?

Reply:
I am not using it. I had in mind someone might want to use the structures in the class to determine intersects on another side of a non-conforming interface without an additional search. However, the same information can also be determined from the vector of requests given to ```distributed_compute_...``` which makes to possible to drop the template parameter, so I will delete it :)

Reply:
I removed the paremter in [c42af30](https://github.com/dealii/dealii/pull/15165/commits/c42af30ce243fe3bbb7b191d59bba9e5a2be3b52)